commercedbsync/src/com/sap/cx/boosters/commercedbsync/repository/platform/MigrationHybrisMSSqlPlatform.java

```diff
@@ -96,7 +101,139 @@ public HybrisMSSqlModelReader(Platform platform) {
+                }
+
+                Table table = superReadTable(metaData, values);
+                if (table != null) {
+                    determineAutoIncrementFromResultSetMetaData(table, table.getColumns());
+                    int idx = 0;
+
+                    while(true) {
+                        while(idx < table.getIndexCount()) {
+                            Index index = table.getIndex(idx);
+                            if (index.isUnique() && this.existsPKWithName(metaData, table, index.getName())) {
+                                table.removeIndex(idx);
+                            } else {
+                                ++idx;
+                            }
+                        }
+
+                        return table;
+                    }
+                } else {
+                    return table;
+                }
+            }
+        }
+
+        private Table superReadTable(DatabaseMetaDataWrapper metaData, Map values) throws SQLException {
+            String tableName = (String)values.get("TABLE_NAME");
+            Table table = null;
+            if (tableName != null && tableName.length() > 0) {
+                table = new Table();
+                table.setName(tableName);
+                table.setType((String)values.get("TABLE_TYPE"));
+                table.setCatalog((String)values.get("TABLE_CAT"));
+                table.setSchema((String)values.get("TABLE_SCHEM"));
+                table.setDescription((String)values.get("REMARKS"));
+                table.addColumns(this.readColumns(metaData, tableName));
+                table.addForeignKeys(this.readForeignKeys(metaData, tableName));
+                table.addIndices(this.readIndices(metaData, tableName));
+                Collection primaryKeys = this.readPrimaryKeyNames(metaData, tableName);
+                Iterator it = primaryKeys.iterator();
+
+                while(it.hasNext()) {
+                    table.findColumn((String)it.next(), true).setPrimaryKey(true);
+                }
+
+                if (this.getPlatformInfo().isSystemIndicesReturned()) {
+                    this.removeSystemIndices(metaData, table);
+                }
+            }
+
+            return table;
+        }
+
+        private boolean existsPKWithName(DatabaseMetaDataWrapper metaData, Table table, String name) {
+            try {
+                ResultSet pks = metaData.getPrimaryKeys(table.getName());
+                boolean found = false;
+
+                while(pks.next() && !found) {
+                    if (name.equals(pks.getString("PK_NAME"))) {
+                        found = true;
+                    }
+                }
+
+                pks.close();
+                return found;
+            } catch (SQLException var6) {
+                throw new DdlUtilsException(var6);
+            }
+        }
+
+        protected void determineAutoIncrementFromResultSetMetaData(Table table, Column[] columnsToCheck) throws SQLException {
```

Comment:
What is the purpose of this method? SAP Commerce is not utilizing MSSQL autoincrement (`IDENTITY`) functionality.